core/src/main/java/com/linecorp/armeria/common/ConnectionEventKey.java

```diff
@@ -0,0 +1,55 @@
+package com.linecorp.armeria.common;
+
+import java.net.InetSocketAddress;
+
+import com.linecorp.armeria.common.annotation.Nullable;
+
+import io.netty.util.AttributeKey;
+
+public class ConnectionEventKey {
+    public static final AttributeKey<ConnectionEventKey> CONNECTION_EVENT_KEY =
+            AttributeKey.valueOf(ConnectionEventKey.class, "CONNECTION_EVENT_KEY");
+    private final InetSocketAddress remoteAddress;
+    private final InetSocketAddress localAddress;
+    @Nullable
+    private SessionProtocol protocol;
+
+    @Nullable
+    private SessionProtocol desiredProtocol;
+
```

Comment:
```suggestion
```
because this isn't used.

Reply:
I forgot to use it. 😆 Thank you for the review. 

This field is going to be used for identifying pending connection events on the client side.

Suppose you count a connection pending event (Like the ConnectionPoolMetrics class) and want to decrease the count after the connection is opened. In that case, you need a key to identify the previous connection pending event. Since the final protocol is undetermined in the connection pending phase,  (desiredProtocol, remoteAddress, localAddress) can be used as the connection event key.

```
    void connectionPending(SessionProtocol desiredProtocol,
                           InetSocketAddress remoteAddress,
                           InetSocketAddress localAddress,
                           AttributeMap attrs) throws Exception;

    void connectionOpened(@Nullable SessionProtocol desiredProtocol,
                          SessionProtocol protocol,
                          InetSocketAddress remoteAddress,
                          InetSocketAddress localAddress,
                          AttributeMap attrs) throws Exception;
```